components/scrubber/scrubber.go

```diff
@@ -166,17 +187,43 @@ func (s *structScrubber) StructField(field reflect.StructField, val reflect.Valu
 		}
 		if setExplicitValue {
 			val.SetString(explicitValue)
-			return nil
+		} else {
+			val.SetString(s.Parent.KeyValue(field.Name, val.String()))
 		}
-		val.SetString(s.Parent.KeyValue(field.Name, val.String()))
-		return nil
+		return reflectwalk.SkipEntry
 	}
 
-	return reflectwalk.Walk(val.Interface(), s)
```

Comment:
(note) reflectwalk is always recursive already, there is no need to call it again